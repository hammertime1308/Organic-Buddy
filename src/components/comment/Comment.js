import React from 'react';
import { View, Text } from 'react-native';

export const Comment = ({ comment, firstName, lastName, timestamp }) => {
  return (
    <View
      style={{
        marginLeft: 5,
        marginRight: 5,
        marginBottom: 10,
        backgroundColor: 'rgb(208,208,208)',
        borderRadius: 5,
        elevation: 2,
      }}>
      <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
        <View style={{ flexDirection: 'row', padding: 10 }}>
          <Text
            style={{
              paddingRight: 5,
              fontSize: 10,
              fontWeight: 'bold',
              color: 'gray',
            }}>
            By: {firstName}
          </Text>
          <Text style={{ fontSize: 10, fontWeight: 'bold', color: 'gray' }}>
            {lastName}
          </Text>
        </View>
        <Text style={{ alignSelf: 'center', fontSize: 10, paddingRight: 10 }}>
          {timestamp}
        </Text>
      </View>
      <Text
        style={{
          fontSize: 15,
          paddingLeft: 10,
          paddingRight: 10,
          paddingBottom: 10,
          fontWeight: '700',
        }}>
        {comment}
      </Text>
    </View>
  );
};
