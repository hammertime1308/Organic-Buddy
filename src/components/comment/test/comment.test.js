import React from 'react';
import renderer from 'react-test-renderer';
import { Comment } from '../Comment';

describe('Comment', () => {
  it('Should render comment', () => {
    const tree = renderer
      .create(
        <Comment
          comment={'Sample comment for testing purpose'}
          firstName={'Mandar'}
          lastName={'Kulkarni'}
          timestamp={'2021-06-12T10:17:52Z'}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
