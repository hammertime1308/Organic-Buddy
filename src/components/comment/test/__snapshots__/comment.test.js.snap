// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Should render comment 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(208,208,208)",
      "borderRadius": 5,
      "elevation": 2,
      "marginBottom": 10,
      "marginLeft": 5,
      "marginRight": 5,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
            "fontWeight": "bold",
            "paddingRight": 5,
          }
        }
      >
        By: 
        Mandar
      </Text>
      <Text
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
            "fontWeight": "bold",
          }
        }
      >
        Kulkarni
      </Text>
    </View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 10,
          "paddingRight": 10,
        }
      }
    >
      2021-06-12T10:17:52Z
    </Text>
  </View>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "700",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    Sample comment for testing purpose
  </Text>
</View>
`;
